src/components/OrderPanel/BookingTimeForm/FieldDateAndTimeInput.js

```diff
@@ -234,11 +236,110 @@ const getAllTimeValues = (
     ? new Date(findNextBoundary(startTimeAsDate, 'hour', timeZone).getTime() - 1)
     : null;
 
+  const selectedEndTimeAsDateObject = selectedEndTime ? timestampToDate(selectedEndTime) : null;
+
   const selectedTimeSlot = timeSlots.find(t =>
     isInRange(startTimeAsDate, t.attributes.start, t.attributes.end)
   );
 
-  const endTimes = getAvailableEndTimes(intl, timeZone, startTime, endDate, selectedTimeSlot);
+  const selectedTimeSlotIndex = timeSlots.findIndex(t =>
+    isInRange(startTimeAsDate, t.attributes.start, t.attributes.end)
+  );
+
+  const findLastAdjacent = index => {
+    const current = timeSlots[index];
+    const next = timeSlots[index + 1];
+    return next && isSameDate(current.attributes.end, next.attributes.start)
+      ? findLastAdjacent(index + 1)
+      : index;
+  };
+
+  const findFirstAdjacent = index => {
+    const current = timeSlots[index];
+    const previous = timeSlots[index - 1];
+    return previous && isSameDate(current.attributes.start, previous.attributes.end)
+      ? findFirstAdjacent(index - 1)
+      : index;
+  };
+
+  const combineTimeSlots = (currentTimeSlotIndex, timeSlots, availabilityType) => {
+    if (!timeSlots) {
+      return null;
+    }
+    if (timeSlots.length <= 1 || availabilityType !== 'multipleSeats') {
+      return timeSlots[0];
+    }
+
+    const lastIndex = findLastAdjacent(currentTimeSlotIndex);
+    const firstIndex = findFirstAdjacent(currentTimeSlotIndex);
+
+    const combinedTimeSlot = {
+      ...timeSlots[currentTimeSlotIndex],
+      attributes: {
+        ...timeSlots[currentTimeSlotIndex].attributes,
+        start: timeSlots[firstIndex].attributes.start,
+        end: timeSlots[lastIndex].attributes.end,
+      },
```

Comment:
Perhaps you could add the `seats: smallestSeats` already to this combinedTimeSlot?